vnext/ReactUWP/Views/ScrollViewManager.cpp

```diff
@@ -28,6 +15,268 @@ enum class ScrollViewCommands
+      if (valid)
+      {
+        scrollViewer.HorizontalScrollBarVisibility(showsHorizontalScrollIndicator ? winrt::ScrollBarVisibility::Visible : winrt::ScrollBarVisibility::Hidden);
+      }
+    }
+    else if (propertyName == "showsVerticalScrollIndicator")
+    {
+      const auto [valid, showsVerticalScrollIndicator] = getPropertyAndValidity(propertyValue, true);
+      if (valid)
+      {
+        scrollViewer.VerticalScrollBarVisibility(showsVerticalScrollIndicator ? winrt::ScrollBarVisibility::Visible : winrt::ScrollBarVisibility::Hidden);
+      }
+    }
+    else if (propertyName == "minimumZoomScale")
+    {
+      const auto [valid, minimumZoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.MinZoomFactor(static_cast<float>(minimumZoomScale));
+      }
+    }
+    else if (propertyName == "maximumZoomScale")
+    {
+      const auto [valid, maximumZoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.MaxZoomFactor(static_cast<float>(maximumZoomScale));
+      }
+    }
+    else if (propertyName == "zoomScale")
+    {
+      const auto [valid, zoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.ChangeView(nullptr, nullptr, static_cast<float>(zoomScale));
+      }
+    }
+    else if (propertyName == "snapToInterval")
+    {
+      const auto [valid, snapToInterval] = getPropertyAndValidity(propertyValue, 0.0);
+      if (valid)
+      {
+        ScrollViewUWPImplementation(scrollViewer).SnapToInterval(static_cast<float>(snapToInterval));
+      }
+    }
+    else if (propertyName == "snapToOffsets")
+    {
+      const auto [valid, snapToOffsets] = getPropertyAndValidity(propertyValue, winrt::single_threaded_vector<float>());
+      if (valid)
+      {
+        ScrollViewUWPImplementation(scrollViewer).SnapToOffsets(snapToOffsets.GetView());
+      }
+    }
+    else if (propertyName == "snapToAlignment")
+    {
+      const auto [valid, snapToAlignment] = getPropertyAndValidity(propertyValue, winrt::SnapPointsAlignment::Near);
+      if (valid)
+      {
+        scrollViewer.HorizontalSnapPointsAlignment(snapToAlignment);
+        scrollViewer.VerticalSnapPointsAlignment(snapToAlignment);
+      }
+    }
+  }
+
+  Super::updateProperties(std::move(reactDiffMap));
+  m_updating = false;
+}
+
+template <typename T>
+std::tuple<bool, T> ScrollViewShadowNode::getPropertyAndValidity(folly::dynamic propertyValue, T defaultValue)
+{
+  if (propertyValue.isNull())
+  {
+    return std::make_tuple(true, defaultValue);
+  }
+
+  if constexpr (std::is_same<T, bool>())
+  {
+    if (propertyValue.isBool())
+    {
+      return std::make_tuple(true, propertyValue.asBool());
+    }
+  }
+
+  if constexpr (std::is_same<T, double>())
+  {
+    if (propertyValue.isDouble())
+    {
+      return std::make_tuple(true, static_cast<T>(propertyValue.asDouble()));
```

Comment:
>asDouble [](start = 64, length = 8)

if you've validated the type is a double, use getDouble instead of asDouble.   as* is a more expensive API that supports conversion, when the types are the same its 'cheap' but still not free because it creates local stack variables including a std::string   (I have another change to clean existing cases of this to send soon)

Reply:
Cool, thanks!